### DISCUSSION TITLE
Trying to install as instructed on Raspberry Pi OS

### DISCUSSION THREAD

**Discussion by steinersupport** [karlcswanson/micboard#42] [General]
I run through the commands as directed. The first thing that fails outright is "npm install --only=prod". If I install npm outright it goes much further but then ultimately fails. Can anyone point me in the right direction?

**Comment by steinersupport** [comment]
It looks like some required steps for Raspberry Pi OS are missing from the instructions provided. While troubleshooting we might have taken a few steps that were unnecessary for this to work so there might be a bit extra. Following is what worked for us in case it helps someone else out. It is pretty raw since it was documented just for my reference. 


Install dependancies
```sudo apt update```
`sudo apt-get install git python3-pip`
`curl -sL https://deb.nodesource.com/setup_10.x | sudo -E bash -`
`sudo apt-get install -y nodejs`
`sudo apt-get install gcc g++ make`
`sudo apt autoremove`
`sudo apt-get update && sudo apt-get install yarn`
`sudo apt install npm`
`sudo apt install python2`
`wget https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.1/install.sh`
`bash install.sh`
`source ~/.bashrc`
`command -v nvm`
`nvm install v10.24.1`

Download micboard
`git clone https://github.com/karlcswanson/micboard.git`

Install micboard software dependencies via npm and pip
`cd micboard/`
`npm install --only=prod`
`pip3 install -r py/requirements.txt`

build the micboard frontend and run micboard
`npm run build`
`python3 py/micboard.py`

Edit `User` and `WorkingDirectory` within `micboard.service` to match your installation. 
`nano micboard.service`
Example Settings:
_____________________________________________

```
[Unit]
Description=Micboard Service
After=network.target

[Service]
Environment=MICBOARD_PORT=8058
ExecStart=/usr/bin/python3 -u py/micboard.py
WorkingDirectory=/home/[username]/micboard
StandardOutput=inherit
StandardError=inherit
Restart=always
User=[username]
AmbientCapabilities=CAP_NET_BIND_SERVICE

[Install]
WantedBy=multi-user.target
```
_____________________________________________

Install it as a service.
`sudo cp micboard.service /etc/systemd/system/`
`sudo systemctl start micboard.service`
`sudo systemctl enable micboard.service`

**Reply by TreyCarey** [reply]
This worked perfectly for me after having issues on Pi as well, thank you!
